pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SnapshotSegmentAbortedTxnProcessorImpl.java

```diff
@@ -0,0 +1,758 @@
+                snapshotSegmentsWriterFuture;
+        private final CompletableFuture<SystemTopicClient.Writer<TransactionBufferSnapshotIndexes>>
+                snapshotIndexWriterFuture;
+
+        private enum OperationState {
+            None,
+            Operating,
+            Closed
+        }
+        private static final AtomicReferenceFieldUpdater<PersistentWorker, PersistentWorker.OperationState>
+                STATE_UPDATER = AtomicReferenceFieldUpdater.newUpdater(PersistentWorker.class,
+                PersistentWorker.OperationState.class, "operationState");
+
+        public enum OperationType {
+            UpdateIndex,
+            WriteSegment,
+            DeleteSegment,
+            Clear
+        }
+
+        private volatile OperationState operationState = OperationState.None;
+
+        ConcurrentLinkedDeque<Pair<OperationType, Pair<CompletableFuture<Void>,
+                Supplier<CompletableFuture<Void>>>>> taskQueue = new ConcurrentLinkedDeque<>();
+
+        public PersistentWorker(PersistentTopic topic) {
+            this.topic = topic;
+            this.snapshotSegmentsWriterFuture =  this.topic.getBrokerService().getPulsar()
+                    .getTransactionBufferSnapshotServiceFactory()
+                    .getTxnBufferSnapshotSegmentService().createWriter(TopicName.get(topic.getName()))
+                    .exceptionally(ex -> {
+                        log.error("{} Failed to create snapshot index writer", topic.getName());
+                        topic.close();
+                        return null;
+                    });
+            this.snapshotIndexWriterFuture =  this.topic.getBrokerService().getPulsar()
+                    .getTransactionBufferSnapshotServiceFactory()
+                    .getTxnBufferSnapshotIndexService().createWriter(TopicName.get(topic.getName()))
+                    .exceptionally((ex) -> {
+                        log.error("{} Failed to create snapshot writer", topic.getName());
+                        topic.close();
+                        return null;
+                    });
+        }
+
+        public CompletableFuture<Void> appendTask(OperationType operationType,
+                                                  Supplier<CompletableFuture<Void>> task) {
+            CompletableFuture<Void> taskExecutedResult = new CompletableFuture<>();
+            switch (operationType) {
+                case UpdateIndex -> {
+                    /*
+                      The update index operation can be canceled when the task queue is not empty,
+                      so it should be executed immediately instead of appending to the task queue.
+                      If the taskQueue is not empty, the worker will execute the tasks in the queue.
+                     */
+                    if (!taskQueue.isEmpty()) {
+                        executeTask();
+                        return CompletableFuture.completedFuture(null);
```

Comment:
The operation of updating indexes has yet to be complete when this future is complete, right?

Reply:
The operation of updating indexes is meaningless when there are other operations in the queue. So just completing it is enough.

Reply:
##### Additional question

The operation `update indexes` will do three things:
- update indexes
- update max read position
- update unsealed aborts 

But the operation `update indexes` followed by `take new segment` only do two things:
- update indexes
- update max read position

So is it okay to skip task `update unsealed aborts`?